libs/content/workspace-ffi/src/apple/common_ffi.rs

```diff
@@ -134,15 +135,41 @@ pub unsafe extern "C" fn clipboard_paste(obj: *mut c_void, content: *const c_cha
 /// # Safety
 /// obj must be a valid pointer to WgpuEditor
 #[no_mangle]
-pub unsafe extern "C" fn clipboard_paste_image(
-    obj: *mut c_void, content: *const u8, length: usize,
+pub unsafe extern "C" fn clipboard_send_image(
+    obj: *mut c_void, content: *const u8, length: usize, is_paste: bool,
 ) {
     let obj = &mut *(obj as *mut WgpuWorkspace);
-    let content = std::slice::from_raw_parts(content, length).to_vec();
-    obj.context.push_event(workspace_rs::Event::Paste {
-        content: vec![ClipContent::Png(content)],
-        position: egui::Pos2::ZERO, // todo: cursor position
-    });
+    let img = std::slice::from_raw_parts(content, length).to_vec();
+    let content = vec![ClipContent::Png(img)];
+    let position = egui::Pos2::ZERO; // todo: cursor position
```

Comment:
At some point we'll support passing this in so we can put the image where you dropped it instead of at the cursor or whatever. In the meantime, I thought macos had behavior where if you drop an image into the file tree it imports it as a file and if you drop it into the editor it additionally inserts a link - is that right, and if so was that behavior preserved?

Reply:
That behavior is still preserved. The difference is that if you drop an image into the file tree, it goes through entirely separate logic, and it uses the import functions from core.